### PR TITLE
Clean up engine hardcodes.

### PR DESCRIPTION
**Pull request by oliverchang**
- Have a single place for defining the list of engines.
- Remove "blackbox" engine as it's unused and unfinished.
- Remove builtin_fuzzers.

### CODE REVIEW

**Review by jonathanmetzman — APPROVED** [review]
LGTM

File: src/clusterfuzz/_internal/bot/fuzzers/init.py [hunk 1]
```diff
@@ -13,24 +13,23 @@
...
+  else:
+    engines = fuzzing.PUBLIC_ENGINES
+
+  for engine_name in engines:
+    mod = importlib.import_module(
+        f'clusterfuzz._internal.bot.fuzzers.{engine_name}.engine')
+
+    engine.register(engine_name, mod.Engine)
+    if include_lowercase and engine_name.lower() != engine_name:
+      engine.register(engine_name.lower(), mod.Engine)
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply src/clusterfuzz/_internal/bot/fuzzers/init.py:35]: Yes, see line 33 which registers the name as-is.

File: src/clusterfuzz/_internal/fuzzing/__init__.py [hunk 2]
```diff
@@ -11,3 +11,15 @@
...
+"""Fuzzing module."""
+
+PUBLIC_ENGINES = (
+    'libFuzzer',
+    'afl',
+    'honggfuzz',
+    'googlefuzztest',
+)
+
+PRIVATE_ENGINES = ('syzkaller',)
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply src/clusterfuzz/_internal/fuzzing/__init__.py:23]: Our syzkaller integration is highly android/internal specific and most likely will not work generally.

File: src/clusterfuzz/_internal/bot/tasks/minimize_task.py [hunk 3]
```diff
@@ -27,7 +27,7 @@
 from clusterfuzz._internal.base import utils
 from clusterfuzz._internal.bot import testcase_manager
 from clusterfuzz._internal.bot.fuzzers import engine_common
-from clusterfuzz._internal.bot.fuzzers.libFuzzer.engine import LibFuzzerEngine
+from clusterfuzz._internal.bot.fuzzers.libFuzzer.engine import Engine
```
[the remark this answers is not in the document]
Comment (oliverchang) [reply src/clusterfuzz/_internal/bot/tasks/minimize_task.py]: Good point, fixed!

**Pull request merged**